src/actions/useTurnInto.actions.tsx

```diff
@@ -10,20 +11,39 @@ export default function useTurnInto() {
   const selectedElements = rnbw.elements.getSelectedElements();
 
   function turnInto(tagName: string) {
-    selectedElements.forEach((uid) => {
-      //replace the tag name
+    const codeViewInstanceModel = rnbw.files.getEditorRef().current?.getModel();
+    if (!codeViewInstanceModel) return;
+    helperModel.setValue(codeViewInstanceModel.getValue());
+
+    selectedElements.forEach(async (uid) => {
       const node = rnbw.elements.getElement(uid);
-      if (node) {
-        const { startCol, startLine, endCol, endLine } =
-          node.data.sourceCodeLocation;
-
-        const edit = {
-          range: new Range(startLine, startCol, endLine, endCol),
-          text: `<${tagName}>${node.data.content}</${tagName}>`,
-        };
-        helperModel.applyEdits([edit]);
-      }
+      if (!node) return;
+      const nodeAttribute = rnbw.elements.getElementSettings(uid);
+
+      // Getting existing tag attributes
+      const attributeCode = Object.entries(nodeAttribute)
+        .map(([attrName, attrValue]) => `${attrName}="${attrValue}"`)
+        .join(" ");
+
+      const { startTag, endTag, startLine, startCol, endLine, endCol } =
+        node.data.sourceCodeLocation;
+      if (!startTag || !endTag) return;
+      const text = codeViewInstanceModel.getValueInRange(
+        new Range(
+          startTag.endLine,
+          startTag.endCol,
+          endTag.startLine,
+          endTag.startCol,
+        ),
+      );
+      const edit = {
+        range: new Range(startLine, startCol, endLine, endCol),
+        text: `<${tagName} ${attributeCode}>${text}</${tagName}>`,
+      };
+      helperModel.applyEdits([edit]);
     });
+    const code = html_beautify(helperModel.getValue());
```

Comment:
Let's not prettify entire code. We only prettify or format the code that is added or edited by rnbw.
And we either use prettier or html_beautify. Let's not use both.

We can eliminate it slowly. By replacing html_beautify with preetier.

Reply:
Fixed